plugins/node/opentelemetry-instrumentation-nestjs-core/src/index.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Instrumentation } from './instrumentation';
+
+export * from './instrumentation';
+export default Instrumentation;
```

Comment:
why do we add a default export here?

Reply:
Why not? It's essentially one-class-package, so the default doesn't have any "competition".
In that way instrumentation can be just exported more conveniently without knowing the name I chose for the instrumentation class. Dynamic loading (if that will ever be a thing. Currently for many reasons, it's not obvious why it should) would be easier and it's just a small improvement on convenience in some cases. Interested in your thoughts though.

Reply:
I don't care that much if it is done here or not.
I was just wondering as I haven't seen this at other places. Don't think there is a single export default in API or Core packages.

Reply:
Yeah... And for API or Core, I understand why - they are kind of bundles of many useful blocks.

Reply:
I would prefer to do what the other instr packages do for the sake of consistency. If we want to change it then it should be changed for all of them in another PR.

Reply:
Even though it's additive - the "normal" way stays the same? For what it's worth I've added the default export before to instrumentations: Knex, Mysql2..

Reply:
@dyladan, is this holding the merge back?

Reply:
If a package only exports 1 thing (and only ever will) then `default` make sense. But if something exports multiple keys as in this case, then (my preference) is to not use `default` mainly from an end user usage perspective.

Reply:
I can be convinced, but I'd like a more compelling argument for "the end user usage perspective". Even if the instrumentation class is not strictly the only thing, it's **always** the **main** thing, don't you think?

Let me explain:

The vast majority of instrumentation packages will export one thing: The instrumentation class. Everyone hopefully knows how to use it - you can pass it to "registerInstrumentation" call to enable it, and call it a day.

With default export, there's one thing you need to know: the package name(which you needed to install it).
Without default export, you need to know **twice**(a little dramatization :wink:) as many things: the package name(to import) **and** the name that was given to the Instrumentation class inside the package(`NestInstrumentation`).

Some instrumentations also have types and enums that are exported but importing those is (hopefully - since we do want to default to the most useful configuration) an edge case.

Reply:
> it's always the main thing, don't you think?

While its true it isn't the elephant in the room IMO, i agree with @dyladan that we have done that since the opencensus days and if we want this to change, this would need be changed everywhere and not having specific package that does this and other use a named export.
It's a problem for me at least for consistency sake

Reply:
Just to be clear - I wasn't suggesting removing named import, but adding a default one for convenience.
The rule of "always have default export"(that's how I took your premise) cannot and should not be enforced generally because there are cases where it is not applicable so chasing that consistency - is that a good reason to never use it? I don't think so.

But the general sentiment from you is clear and I respect that - removed the default export.